File: Chapter 05/JS/section_5_6_crypto_tracker/after/core/ports/priceProviderPort.js
/**
 * PORT – Defines "What" we need (lives in Core).
 * Using a base class to act as the interface contract.
 */
class PriceProviderPort {
    async getBitcoinPrice() {
        throw new Error("Method 'getBitcoinPrice()' must be implemented by the Adapter.");
    }
}

module.exports = PriceProviderPort;

File: Chapter 05/Python/section_5_6_crypto_tracker/after/core/ports/price_provider_port.py
from abc import ABC, abstractmethod

class PriceProviderPort(ABC):
    """PRIMARY PORT (Driven).
    
    Defines 'What' the domain needs from the outside world.
    The Abstract Base Class enforces the contract strictly, ensuring 
    any infrastructure adapter is compatible with the Core.
    """
    
    @abstractmethod
    def get_bitcoin_price(self) -> float:
        """Retrieves the current market price of Bitcoin.
        
        Returns:
            float: The price of 1 BTC in USD.
        """
        pass